src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c

```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+                            break;
+                    }
+            }
+            break;
+    }
+    func_8013E1C8(&this->skelAnime, D_809CDC7C, 2, &this->animationIndex);
+    this->actionFunc = func_809CD328;
+}
+
+void func_809CD328(EnBji01* this, GlobalContext* globalCtx) {
+
+    switch (func_80152498(&globalCtx->msgCtx)) {
+        case 0:
+            Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x444);
+            func_809CCDE0(this, globalCtx);
+            if (this->actor.shape.rot.y == this->actor.yawTowardsPlayer) {
+                func_801518B0(globalCtx, this->textId, &this->actor);
+            }
+            break;
+        case 4:
+            if (func_80147624(globalCtx) != 0) {
+                this->actor.flags &= 0xFFFEFFFF;
+                this->actor.params = 1;
+                switch (globalCtx->msgCtx.choiceIndex) {
+                    case 0:
+                        func_8019F208();
+                        func_801477B4(globalCtx);
+                        func_809CD634(this, globalCtx);
+                        break;
+                    case 1:
+                        func_8019F230();
+                        switch (gSaveContext.playerForm) {
+                            case PLAYER_FORM_DEKU:
+                                func_80151938(globalCtx, 0x5F0U);
+                                break;
+                            case PLAYER_FORM_HUMAN:
+                                func_80151938(globalCtx, 0x5F8U);
+                                break;
+                            case PLAYER_FORM_GORON:
+                            case PLAYER_FORM_ZORA:
+                                func_80151938(globalCtx, 0x5E1U);
+                                break;
+                        }
+                        break;
+                }
+            }
+            break;
+        case 5:
+            if (func_80147624(globalCtx) != 0) {
+                this->actor.flags &= 0xFFFEFFFF;
+                switch (globalCtx->msgCtx.unk11F04) {
+                    case 1502:
+                        func_8013E1C8(&this->skelAnime, D_809CDC7C, 3, &this->animationIndex);
+                        func_80151938(globalCtx, 0x5DFU);
+                        break;
+                    case 1508:
+                        func_80151938(globalCtx, 0x5E7U);
+                        break;
+                    case 1509:
+                        func_80151938(globalCtx, 0x5E0U);
+                        break;
+                    case 1511:
+                        func_80151938(globalCtx, 0x5E5U);
+                        break;
+                    case 1500:
+                    case 1501:
+                    case 1503:
+                    case 1516:
+                    case 1517:
+                    case 1518:
+                    case 1522:
+                    case 1525:
+                        func_80151938(globalCtx, (globalCtx->msgCtx.unk11F04 + 1) & 0xFFFF);
```

Comment:
Would this match?
```suggestion
                        func_80151938(globalCtx, (u16)(globalCtx->msgCtx.unk11F04 + 1));
```

Reply:
Or maybe the prototype of this function is wrong. I assume something is needed?

Reply:
& with 0xFFFF does match. In fact it's what the raw mips_to_c produced, but casting to u16 also matched, and I thought it was clearer, because that function takes a u16 as an argument, and &ing with 0xFFFF could be either s16 or u16.

Reply:
actually, I thought this was the hours to moon crash calculation which looks very similar. I have never looked deeply at this function so I'm not sure.

Reply:
`& 0xFFFF` is not equivalent to an `(s16)` cast. IDO makes shifts operations to cast to signed types.

Reply:
`func_80151938` takes a u16 anyway, so it should match without.

Reply:
I assume this will match without either u16 or &0xFFFF but I'm rebuilding so that I can run diff.py before I commit.

Reply:
It does match without either, so I'll be pushing that change manually.

Reply:
Will be fixed in a batch commit.